### DISCUSSION TITLE
getServerSideProps to all components

### DISCUSSION THREAD

**Discussion by daviavmello** [vercel/next.js#22198] [Help]
Is it possible to pass the props from `getServerSideProps` to more than one component without replicating the code?

**Comment by jamesmosier** [comment]
There is currently no global way to do this. `getServerSideProps` will only return the props to the `page` component that implements it.

If you have multiple pages who have the same logic you could break out the `getServerSideProps` inner logic to another function that you could reuse.

Or if you are wanted to pass props from the page down to the other components within that page you could use something like the [Context API](https://reactjs.org/docs/context.html#when-to-use-context) in that page component ([or this example](https://github.com/vercel/next.js/tree/canary/examples/with-context-api)).

**Reply by yogeshvishnole** [reply]
looking for a very reusable and clean solution for it.

**Reply by Vardiak** [reply]
This is just bad design if we don't want data to be refetched at each page transition + server side rendering...

**Reply by yogeshvishnole** [reply]
Hello vardiak , Is using next-pipe-props is good idea. You can see it here https://www.npmjs.com/package/next-pipe-props
https://frontend-devops.com/blog/pipe-serverside-props-in-nextjs

**Reply by Vardiak** [reply]
@yakovlevyuri If I understand correctly, it would still do the request at each page instead of one time at app loading.

**Reply by stephanrotolante** [reply]
@Vardiak I don't think it would be bad design if you are authenticating the user to make sure they have access

**Reply by stephanrotolante** [reply]
@daviavmello
 I don't know what you are trying to do but I think this could be done. Because all that nextjs is doing it exporting a function named `getServerSideProps` from the page. All that function does is return JSON. Maybe you can create reusable function that does the default operations that you need, plus takes in a call back function to execute any extra logic that you need
